File: Restaurant-information-system/src/main/java/com/galaxy/Restaurantinformationsystem/Repository/MenuRepository.java
package com.galaxy.Restaurantinformationsystem.Repository;

import com.galaxy.Restaurantinformationsystem.Entity.MenuEntity;
import com.galaxy.Restaurantinformationsystem.Entity.StoreEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;

@Repository
public interface MenuRepository extends JpaRepository<MenuEntity, Long> {

    @Query("select m from MenuEntity m where m.store.id = ?1")
    ArrayList<MenuEntity> findByStore_SPK(Long id);

    List<MenuEntity> deleteByStore(StoreEntity store);

}

File: src/main/java/com/galaxy/restaurant/information/system/restaurant/repository/MenuRepository.kt
package com.galaxy.restaurant.information.system.restaurant.repository

import com.galaxy.restaurant.information.system.restaurant.entity.Menu
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MenuRepository : JpaRepository<Menu, Long> {

    /**
     * 가게 id로 메뉴 찾기
     *
     * @param storeId 가게 id
     * @return 메뉴 리스트
     */
    fun findByStoreId(storeId: Long): List<Menu>
}
